File: backend/app/models/ageportal.py
from ..db import db
from sqlalchemy import UUID


class Ageportal(db.Model):
    __tablename__ = 'ageportal'
    guid = db.Column(UUID, primary_key=True, nullable=False, unique=True)
    name = db.Column(db.String(120), nullable=False)
    alias = db.Column(db.String(120), nullable=False)
    description = db.Column(db.String(255), nullable=True)
    url = db.Column(db.String(255), nullable=True)
    type = db.Column(db.String(255), nullable=True)
    ishosted = db.Column(db.Boolean, nullable=False)
    status = db.Column(db.String(120), nullable=False)

    def to_dict(self):
        return {
            "guid": self.guid,
            "name": self.name,
            "alias": self.alias,
            "description": self.description,
            "url": self.url,
            "type": self.type,
            "ishosted": self.ishosted,
            "status": self.status
        }

    def __repr__(self):
        return f'AGE Portal {self.name}'

File: frontend/src/models/age-portal.ts
import {AbstractModel} from "./abstract-models.ts";

interface AgePortalProps {
    guid: string;
    name: string;
    alias: string;
    description?: string;
    url?: string;
    type?: string;
    ishosted: boolean;
    status: string;
}

export class AgePortal extends AbstractModel {
    class: string = 'AgePortal';
    guid: string;
    name: string;
    alias: string;
    description?: string;
    url?: string;
    type?: string;
    ishosted: boolean;
    status: string;

    constructor(props: AgePortalProps) {
        super()
        this.guid = props.guid;
        this.name = props.name;
        this.alias = props.alias;
        this.description = props.description;
        this.url = props.url;
        this.type = props.type;
        this.ishosted = props.ishosted;
        this.status = props.status;
    }
}